zeebe/auth/src/main/java/io/camunda/zeebe/auth/impl/JwtAuthorizationDecoder.java

```diff
@@ -91,6 +90,13 @@ public Map<String, Object> decode() {
           decodedJWT.getClaim(Authorization.AUTHORIZED_USER_KEY).asLong());
     }
 
+    claimMap.putAll(
+        decodedJWT.getClaims().entrySet().stream()
+            .filter(entry -> entry.getKey().startsWith(USER_TOKEN_CLAIM_PREFIX))
+            .collect(
+                Collectors.toMap(
+                    Map.Entry::getKey, claimEntry -> claimEntry.getValue().as(Object.class))));
+
```

Comment:
@maryarm @Ben-Sheppard This is for the Mapping rules cases, right? I feel we're going the wrong way by putting this in the JWT. We talked before about we wanted to get rid of this JWT as it provides no value for Zeebe. By putting everything in this JWT now it's going to be difficult to differentiate when we need to check the authorized user key, or when we need to check mapping rules.

I'd very much prefer these to be different [AuthDataFormats](https://github.com/camunda/camunda/blob/main/zeebe/protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/encoding/AuthInfo.java#L97-L106). One specific for user keys and one specific for user tokens claims.

Reply:
Absolutely agree here @remcowesterhoud - the addition of them here is just to keep the tasks moving and have some way to iterate over. The eventual idea (as we spoke about and mentioned [here](https://github.com/camunda/camunda/pull/22853#pullrequestreview-2341212539)) is to not use this JWT.

Reply:
Do we have an issue to refactor this already?

Reply:
I can't see one yet, but I will create an issue for it